src/plugins/kubectlGeneration/generateKubectlCommandPlugin.ts

```diff
@@ -0,0 +1,77 @@
+import * as vscode from "vscode";
+import { getReadySessionProvider } from "../../auth/azureAuth";
+import { failed } from "../../commands/utils/errorable";
+import { getAKSDocsRAGClient } from "./aksDocsRag/client";
+import { getKubectlCommandPluginResponse } from "../shared/pluginResponses";
+import { AgentRequest, ILocalPluginHandler, LocalPluginArgs, LocalPluginEntry, LocalPluginManifest, ResponseForLanguageModelExtended } from "../../types/@azure/AzureAgent";
+
+const generateKubectlCommandFunctionName = "generate_kubectl_command";
+
+export const generateKubectlCommandPluginManifest: LocalPluginManifest = {
+    name: "generateKubectlCommandPlugin",
+    version: "1.0.0",
+    functions: [
+        {
+            name: generateKubectlCommandFunctionName,
+            parameters: [],
+            returnParameter: {
+                type: "object",
+            },
+            willHandleUserResponse: false,
+        }
+    ]
+};
+
+async function handleKubectlCommandGeneration(agentRequest: AgentRequest): Promise<ResponseForLanguageModelExtended> {
+    const sessionProvider = await getReadySessionProvider();
+
+    if (failed(sessionProvider)) {
+        return { responseForLanguageModel: { text: "Failed to get session provider." } };
+    }
+
+    const client = getAKSDocsRAGClient(sessionProvider.result);
+    const promptHistory = agentRequest.context.history.filter(c => Object.hasOwn(c, "prompt")) as vscode.ChatRequestTurn[];
```

Comment:
Proper error handling here please:

```
const promptHistory = agentRequest.context.history.filter(c => Object.hasOwn(c, "prompt")) as vscode.ChatRequestTurn[];
    if (promptHistory.length === 0) {
        return { responseForLanguageModel: { text: "No prompt found in chat history." } };
    }
    const prompt = promptHistory[promptHistory.length - 1].prompt; // get latest prompt from chat history
```

Reply:
This history context was removed in favor of using the `userPrompt` from the agent